cmdstanpy/model.py

```diff
@@ -818,16 +818,19 @@ def generate_quantities(
             sample_csv_files = mcmc_sample.runset.csv_files
             sample_drawset = mcmc_sample.get_drawset()
             chains = mcmc_sample.chains
+            chain_ids = mcmc_sample.chain_ids
         elif isinstance(mcmc_sample, list):
+            if len(mcmc_sample) < 1:
+                raise ValueError('MCMC sample cannot be empty list')
             sample_csv_files = mcmc_sample
+            chains = len(sample_csv_files)
+            chain_ids = [x + 1 for x in range(chains)]
```

Comment:
Does chain_ids affect the results anyway?

Reply:
yes - the id is captured by CmdStan and passed in to all the stan::services methods - e.g. - https://github.com/stan-dev/cmdstan/blob/c709a0927eb191655b29764688ed1b0a75e36a0c/src/cmdstan/command.hpp#L284

this is used to advance the RNG  -
https://github.com/stan-dev/stan/blob/ae35c12b0f93fd8253b72b84fcb26ec91ae8d52a/src/stan/services/diagnose/diagnose.hpp#L29

the chain_id makes a convenient loop index, but really, it's there so that we can extend the notion of "split R-hat" to multiple chains all with the same seed.

that's why this needs to be correctly implemented.  the next thing to do is to write a little python script that can be called via SLURM to show how to run 100 or 1000 chains on a cluster.

Reply:
just check with Bob on what the right thing to do, which is this:

if the user doesn't specify the seed, then the constructor should choose a random seed based on the timestamp (as in CmdStan).